pkg/common/stringset/stringset_test.go

```diff
@@ -97,3 +97,46 @@ func TestStringSet_ToSlice(t *testing.T) {
 		assert.True(t, testSet.Contains(item))
 	}
 }
+
+func TestStringSet_Intersect(t *testing.T) {
+	testSet := &stringSet{
+		m: make(map[string]bool),
+	}
+
+	testSet1 := &stringSet{
+		m: make(map[string]bool),
+	}
+
+	testItems := []string{
+		"testitem1",
+		"testitem2",
+	}
+
+	// Add testItems to testSet
+	for _, item := range testItems {
+		testSet.m[item] = true
+	}
+
+	testItems1 := []string{
+		"testitem1",
+		"testitem2",
+		"testitem3",
+	}
+
+	// Add testItems1 to testSet1
+	for _, item := range testItems1 {
+		testSet1.m[item] = true
+	}
+
+	// Intersect the two stringSet
+	intersection := testSet.Intersect(testSet1)
```

Comment:
Use test table and add more unit tests such as
- No intersection result
- Empty sets
- both set are purely different.

Reply:
ok, done.

Reply:
Use table driven tests, rather than creating separate tests.
https://github.com/golang/go/wiki/TableDrivenTests

Reply:
cool stuff!
Thanks I understand what you meant now.
Tests unified in one according to TableDrivenTests.